tests/arc/tnon_ref_cursor_in_object.nim

```diff
@@ -0,0 +1,49 @@
+discard """
+  targets: "c js vm"
+  description: "Ensure that `.cursor` works for non-ref types when used on object fields"
+"""
+
+type
+  Resource = object
+    value: int
+
+  Object = object
+    r {.cursor.}: Resource
+    s {.cursor.}: seq[Resource]
+
+var numDestroy = 0
+
+proc `=copy`(x: var Resource, y: Resource) {.error.} # disallow full copies
+proc `=destroy`(x: var Resource) =
+  inc numDestroy
+
+proc test() =
+  # perform the test in procedure so that globals aren't used (their different
+  # semantics with regards to destruction would interfere)
+  var
+    r = Resource(value: 1) # initialize a resource
+    s = @[Resource(value: 2)]
+
+  # make sure no copy is required in the initializer expression:
+  var o = Object(r: r, s: s)
+
+  # copying the object doesn't perform a full copy of the cursor fields:
+  var o2 = o
+  discard addr(o2) # prevent `o2` from being turned into a cursor
```

Comment:
apologies for the possibly dumb question, but I'm pretty far behind on cursor inferencing:

does taking the address block it from being a cursor because the analysis assumes taking an `addr` means all bets are off (conservatism)?

Reply:
I don't think it's a dumb question, no worries.

Yep, that's the behaviour of the current implementation. However, relying on `addr` blocking `o2` from being turned into a cursor is a bit brittle -- the better approach is likely to run the test with `--cursorInference:off`.